lib/tidewave/mcp/tools/source.ex

```diff
@@ -157,4 +168,91 @@ defmodule Tidewave.MCP.Tools.Source do
 
     Path.join([lib_or_src | Enum.reverse(in_app)])
   end
+
+  # IEx.Introspection.decompose
+
+  defp decompose(atom, _context) when is_atom(atom), do: atom
+
+  defp decompose({:__aliases__, _, _} = module, context) do
+    Macro.expand(module, context)
+  end
+
+  defp decompose({:/, _, [call, arity]}, context) do
+    case Macro.decompose_call(call) do
+      {_mod, :__info__, []} when arity == 1 ->
+        {Module, :__info__, 1}
+
+      {mod, fun, []} ->
+        {Macro.expand(mod, context), fun, arity}
+
+      {fun, []} ->
+        {find_decompose_fun_arity(fun, arity, context), fun, arity}
+
+      _ ->
+        :error
+    end
+  end
+
+  defp decompose(call, context) do
+    case Macro.decompose_call(call) do
+      {_mod, :__info__, []} ->
+        {Module, :__info__, 1}
+
+      {maybe_sigil, [_, _]} ->
+        case Atom.to_string(maybe_sigil) do
+          "sigil_" <> _ ->
+            {find_decompose_fun_arity(maybe_sigil, 2, context), maybe_sigil, 2}
+
+          _ ->
+            :error
+        end
+
+      {mod, fun, []} ->
+        {Macro.expand(mod, context), fun}
+
+      {fun, []} ->
+        {find_decompose_fun(fun, context), fun}
```

Comment:
I assume we may remove that clause, since we only expect full Module.function references?

Reply:
Yes, agreed.